### PR TITLE
Users bcrpyt

### PR DESCRIPTION
**Pull request by tobara**
-Added flash notifications
-Added Rspec, Capybara , and Database Cleaner (added user_can_sign_up_spec.rb)
-Added validations to User model

User can successfully sign up, at which time they will be redirected to sign-in (root) page
If unsuccessful they will be redirected back to the sign up page, where they will notified to of the message errors from the user model.

**Still need to add password confirmation match validation

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
+  validates :email, :username, presence: true, uniqueness: true
+  validates :first_name, :last_name, presence: :true
+
   has_secure_password
+
+  private
+
+  def validate_password?
+    password.present? || password_confirmation.present?
+  end
```
Comment (kevlarr) [inline app/models/user.rb:11]: Is this being used anywhere?

Comment (tobara) [reply app/models/user.rb]: My intention was to use it, but at the moment no.

Comment (tobara) [reply app/models/user.rb:11]: Not one bit.. removing it next branch.

Comment (kevlarr) [reply app/models/user.rb:11]: I wouldn't have even noticed that if it hadn't been nested under `private` (meaning it had to be used in the class somewhere)

**Pull request merged**